cmd/thanos/flags.go

```diff
@@ -124,3 +125,46 @@ func modelDuration(flags *kingpin.FlagClause) *model.Duration {
 
 	return value
 }
+
+type pathOrContent struct {
+	name string
+
+	path    *string
+	content *string
+}
+
+func (p *pathOrContent) Content() ([]byte, error) {
+	if p.path != nil && p.content != nil {
+		return nil, errors.Errorf("Both file and content are set for %s", p.name)
+	}
+
+	if p.path != nil {
+		c, err := ioutil.ReadFile(*p.path)
+		if err != nil {
+			return nil, errors.Wrap(err, fmt.Sprintf("loading YAML file %s for %s", *p.path, p.name))
+		}
+		return c, nil
+	}
+
+	if p.content != nil {
+		return []byte(*p.content), nil
+	}
+
+	return nil, nil
+}
+
+func regCommonObjStoreFlags(cmd *kingpin.CmdClause, suffix string) *pathOrContent {
+	fileFlagName := fmt.Sprintf("objstore%s.config-file", suffix)
+	bucketConfFile := cmd.Flag(fileFlagName, fmt.Sprintf("Path to YAML file that contains object store%s configuration.", suffix)).
+		PlaceHolder("<bucket.config-yaml-path>").String()
+
+	bucketConf := cmd.Flag(fmt.Sprintf("objstore%s.config", suffix), fmt.Sprintf("Alternative to '%s' flag. Object store%s configuration in YAML.", fileFlagName, suffix)).
+		PlaceHolder("<bucket.config-yaml>").String()
+
+	return &pathOrContent{
+		name: fmt.Sprintf("objstore%s.config", suffix),
+
+		path:    bucketConfFile,
+		content: bucketConf,
```

Comment:
we could just populate `content` here even if a path is passed ... means we can get rid of `Content()` and just have it as a public property.

Reply:
We cannot. (: This is because this flag is not filled in the time of defining it